src/util/simplify_expr_int.cpp

```diff
@@ -885,8 +885,9 @@ bool simplify_exprt::simplify_concatenation(exprt &expr)
       exprt &op=*it;
       if(op.is_true() || op.is_false())
       {
-        bool value=op.is_true();
-        op=constant_exprt(value?ID_1:ID_0, unsignedbv_typet(1));
+        const bool value = op.is_true();
+        op = from_integer(value, unsignedbv_typet(1));
+        result = true;
```

Comment:
I may be confused: there has been a modification, hence the result should be `false`?!

Reply:
Indeed, fixed